linera-chain/src/block.rs

```diff
@@ -17,11 +17,31 @@ use thiserror::Error;
 use crate::{data_types::ExecutedBlock, ChainError};
 
 /// Wrapper around an `ExecutedBlock` that has been validated.
-#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+#[derive(Debug, PartialEq, Eq, Clone)]
 pub struct ValidatedBlock {
     executed_block: Hashed<ExecutedBlock>,
 }
 
+impl Serialize for ValidatedBlock {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        self.executed_block.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for ValidatedBlock {
+    fn deserialize<D>(deserializer: D) -> Result<ValidatedBlock, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        Ok(ValidatedBlock {
+            executed_block: Hashed::deserialize(deserializer)?,
+        })
+    }
+}
```

Comment:
Is this equivalent to `#[serde(transparent)]`?

Reply:
Ah, it is. Thanks for pointing it out.

Reply:
Done in [use serde(transparent)](https://github.com/linera-io/linera-protocol/pull/3076/commits/9387762edffef6c883f7dbaeaae827c39c566c17). 

I also refactored these types to newtype structs.